values/external-dns/external-dns.gotmpl

```diff
@@ -1,20 +1,22 @@
 {{- $v := .Environment.Values }}
 {{- $externalDns := $v | get "charts.external-dns" }}
 {{- $dns := $v.dns }}
-{{- $dnsProvider := first (keys $v.dns) }}
 sources:
 - ingress
 - istio-gateway
 ## - aws, azure, cloudflare, coredns, designate, digitalocoean, google, infoblox, rfc2136
-provider: {{ $dnsProvider }}
-{{- if eq $dnsProvider "google" }}
+
+{{- if hasKey $dns "google" }}
+provider: "google"
 google:
   project: {{ $dns.google.projectId }}
   serviceAccountKey: |
     {{- $dns.google.cloudDnsKey | nindent 4 }}
```

Comment:
can we maybe normalize this and use the property names that are asked by external-dns? Just like for aws and azure? Then we can just do
```yaml
google:  {{- $dns.google | toYaml | nindent 2 }}
```

Reply:
yes, that's good idea

Reply:
done